spec/support/test_http_client_factory.cr

```diff
@@ -0,0 +1,15 @@
+require "./spec_helper"
+
+class TestHttpClientFactory < Awscr::S3::HttpClientFactory
```

Comment:
It sounds like a mock. Is there a naming convention where mocks should start with 'Test'?

Reply:
Would `HttpClientFactoryMock` be better?

Reply:
> Would HttpClientFactoryMock be better?

Sounds good.